src/main/java/hudson/plugins/git/UserMergeOptions.java

```diff
@@ -19,28 +20,60 @@
 
     private String mergeRemote;
     private String mergeTarget;
-    private String mergeStrategy;
+    private MergeCommand.Strategy mergeStrategy;
     private MergeCommand.GitPluginFastForwardMode fastForwardMode;
 
     /**
      * @deprecated use the new constructor that allows to set the fast forward mode.
      */
     @Deprecated
     public UserMergeOptions(String mergeRemote, String mergeTarget, String mergeStrategy) {
-        this(mergeRemote, mergeTarget, mergeStrategy, MergeCommand.GitPluginFastForwardMode.FF);
+        this(mergeTarget);
+        setMergeRemote(mergeRemote);
+        setMergeStrategy(mergeStrategy);
+        setFastForwardMode(MergeCommand.GitPluginFastForwardMode.FF);
+    }
+
+    /**
+     * @Deprecated use the new @DataBoundConstructor where you only need to supply the necessary information.
+     */
+    @Deprecated
+    public UserMergeOptions(String mergeRemote, String mergeTarget, String mergeStrategy, MergeCommand.GitPluginFastForwardMode fastForwardMode) {
+        this(mergeTarget);
+        setMergeRemote(mergeRemote);
+        setMergeStrategy(mergeStrategy);
+        setFastForwardMode(fastForwardMode);
     }
 
     @DataBoundConstructor
-    public UserMergeOptions(String mergeRemote, String mergeTarget, String mergeStrategy,
-            MergeCommand.GitPluginFastForwardMode fastForwardMode) {
-        this.mergeRemote = mergeRemote;
+    public UserMergeOptions(String mergeTarget) {
         this.mergeTarget = mergeTarget;
+    }
+
+    @DataBoundSetter
+    public void setMergeRemote(String mergeRemote) {
+        this.mergeRemote = mergeRemote;
+    }
+
+    @DataBoundSetter
+    public void setMergeStrategy(MergeCommand.Strategy mergeStrategy) {
         this.mergeStrategy = mergeStrategy;
+    }
+
+    @DataBoundSetter
+    public void setFastForwardMode(MergeCommand.GitPluginFastForwardMode fastForwardMode) {
         this.fastForwardMode = fastForwardMode;
     }
 
+    private void setMergeStrategy(String mergeStrategy) {
+        this.mergeStrategy = (mergeStrategy == null ? null : MergeCommand.Strategy.valueOf(mergeStrategy.toUpperCase()));
```

Comment:
Hmm, now I see [this](https://github.com/jenkinsci/git-client-plugin/blob/651205add433440f89679df1bd725a56cb749b92/src/main/java/org/jenkinsci/plugins/gitclient/MergeCommand.java#L39-L42). (Which by the way would have caused mayhem for Turkish and Azeri users trying to use `RECURSIVE`. Always pass `Locale.ENGLISH` unless you really know what you are doing!)

I suspect that changing the type of the `mergeStrategy` field was not settings-compatible; probably it needed to stay a `String` internally, with the getter and setter translating to the enum type with a case conversion. Hence my complaint about `UserMergeOptionsTest`, which is at the wrong level and so will not catch mistakes like this. You need to use `JenkinsRule` and `@LocalData`.